presto-orc/src/main/java/com/facebook/presto/orc/reader/SelectiveStreamReader.java

```diff
@@ -49,5 +52,13 @@ int read(int offset, int[] positions, int positionCount)
      */
     Block getBlock(int[] positions, int positionCount);
 
+    /**
+     * Like getBlock(), but returns a read-only view of the data.
```

Comment:
Wonder under what cases will we use this interface? `Block` is always immutable that implies read-only for `getBlock` as well.

Reply:
Looks like this interface is only used in #13072. Shall we remove it from this PR?

Reply:
nvm, I mess my local branch.

Reply:
`getBlock()` and `getBlockView()` are very similar by nature. Can we instead of asking the callers to decide when is the first load, but doing it internally within each implementation with a flag maybe?

Reply:
@highker The difference between getBlock() and getBlockView() is the following. The caller of `getBlock()` gets a copy of the data (the underlying values arrays) and is guaranteed that this data won't change. `getBlockView()` returns a view into mutable state (values array); the data may change during next call to `read`. Hence, `getBlockView()` is memory and CPU efficient way to get the data that will be looked at "once", e.g. to evaluate a filter function. `getBlock()` must be called when operator is passing data to the next operator which may hold on to that data, e.g. load it into hash table of the join.

With a single `getBlock()` method, the reader cannot tell whether this is the last call that's used to extract a block to pass to the next operator, or an intermediate call used to evaluate a filter function. In case, when a column is only used to evaluate a filter function, there will be no `getBlock()` call at all, e.g. the data will never be passed to the next operator. Hence, it is not possible for the reader to know when it is safe to return a view into the data and when it has to make a hard copy.

Reply:
@mbasmanova @tdcmeehan, thanks for the explanation on `BlockView`. Big fan of the concept of having `BlockLease` or something similar to indicate the "immutable lifecycle" of the view. Having an explicit lock/release phase sounds reasonable. It also helps to distinguish true immutable `Block` vs temporary immutable "views".

Reply:
@tdcmeehan @highker Tim, James, I updated the code to use BlockLease and put the changes into a separate commit. Would you take a look? If looks good, I'll merge this commit into previous one.

Reply:
Looks good to me